src/Plugin/AbstractIframeValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Drupal\stanford_media\Plugin;
+
+use Drupal\stanford_media\Plugin\EmbedValidatorBase;
+
+/**
+ * Abstract IFrame Validator.
+ *
+ */
+abstract class AbstractIframeValidator extends EmbedValidatorBase {
+
+  const EMBEDDOMAIN = '';
+
+  /**
+   * {@inheritDoc}
+   */
+  public function isEmbedCodeAllowed(string $code): bool {
+    $code = str_replace("\n", ' ', $code);
+    preg_match('/<iframe.* src="(.+?)"/', $code, $source_matches);
+    preg_match('/title\s?=\s?"(.*)"/', $code, $title_matches);
```

Comment:
Let's break this down a little bit. I realized that nested html can be parsed with this too.
```suggestion
    preg_match('/<iframe.*?>/', $code, $iframe_code);
    preg_match('/src="(.*?)"/', $iframe_code, $source_matches);
    preg_match('/title="(.*?)"/', $iframe_code, $title_matches);
```